meta-facebook/yv35-gl/src/platform/plat_cpu.c

```diff
@@ -6,11 +6,135 @@
 #include <stdint.h>
 #include <drivers/peci.h>
 #include "intel_peci.h"
+#include "ipmi.h"
+#include "libipmi.h"
 #include "libutil.h"
 #include "hal_peci.h"
 #include <logging/log.h>
+#include <plat_sensor_table.h>
+#include "power_status.h"
 
-LOG_MODULE_REGISTER(pal_cpu);
+LOG_MODULE_REGISTER(plat_cpu);
+
+K_THREAD_STACK_DEFINE(monitor_cpu_stack, MONITOR_CPU_STACK_SIZE);
+struct k_thread monitor_cpu_thread;
+k_tid_t monitor_cpu_tid;
+
+void start_monitor_cpu_thread()
+{
+	LOG_INF("Start thread to monitor CPU");
+
+	monitor_cpu_tid =
+		k_thread_create(&monitor_cpu_thread, monitor_cpu_stack,
+				K_THREAD_STACK_SIZEOF(monitor_cpu_stack), monitor_cpu_handler, NULL,
+				NULL, NULL, CONFIG_MAIN_THREAD_PRIORITY, 0, K_NO_WAIT);
+	k_thread_name_set(&monitor_cpu_thread, "monitor_cpu_thread");
+}
+
+void monitor_cpu_handler()
```

Comment:
From Amithash:

Looks Intel specific and not greatlakes specific. We probably should move it to common and call it monitor_intel_cpu_handler() and start_monitor_cpu_thread() may pick what common handler to use.

Reply:
Hi @jagpalgill ,
Because GL doesn't have PCH, BMC can't get the thermal event.
Thus, BIC needs to monitor to send thermal event SEL to BMC.
Other projects of Intel CPU may not necessarily require this behavior.